sdk/src/types/block/capabilities.rs

```diff
@@ -207,3 +228,163 @@ pub trait CapabilityFlag {
     /// Returns an iterator over all flags.
     fn all() -> Self::Iterator;
 }
+
+#[cfg(feature = "serde")]
+mod serde {
+    use ::serde::{Deserialize, Serialize};
+
+    use super::*;
+
+    impl<Flag> Serialize for Capabilities<Flag> {
+        fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: ::serde::Serializer,
+        {
+            crate::utils::serde::boxed_slice_prefix_hex_bytes::serialize(&self.bytes, serializer)
+        }
+    }
+
+    impl<'de, Flag: CapabilityFlag> Deserialize<'de> for Capabilities<Flag> {
+        fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where
+            D: ::serde::Deserializer<'de>,
+        {
+            Self::from_prefix_box_slice(crate::utils::serde::boxed_slice_prefix_hex_bytes::deserialize(
+                deserializer,
+            )?)
+            .map_err(::serde::de::Error::custom)
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use pretty_assertions::assert_eq;
+
+    use super::*;
+
+    #[derive(Debug)]
+    #[allow(unused)]
+    enum TestFlag {
+        Val1,
+        Val2,
+        Val3,
+        Val4,
+        Val5,
+        Val6,
+        Val7,
+        Val8,
+        Val9,
+    }
+
+    impl TestFlag {
+        const VAL_1: u8 = 0b00000001;
+        const VAL_2: u8 = 0b00000010;
+        const VAL_3: u8 = 0b00000100;
+        const VAL_4: u8 = 0b00001000;
+        const VAL_5: u8 = 0b00010000;
+        const VAL_6: u8 = 0b00100000;
+        const VAL_7: u8 = 0b01000000;
+        const VAL_8: u8 = 0b10000000;
+        const VAL_9: u8 = 0b00000001;
```

Comment:
Why is this the same as VAL_1?

Reply:
Because that's how the flags work, they have to be one of the 8 possible bits, and each is assigned to an index in the byte array

Reply:
Ah in index() it's used different, thanks